### ISSUE TITLE
Better app init cross-region experience

### ISSUE THREAD

**Issue by iamhopaul123** [aws/copilot-cli#1887]
Right now all the IAM roles we created in copilot are hardcoded. This causes a problem that users cannot create an app with the same name using different profile region, since IAM roles are global.

**Comment by kohidave** [comment]
There's an issue for this somewhere about providing better names - but the global app name restriction was intentional. Having multiple same names apps, I think, would be just too confusing. I think instead we should provide better and earlier error messages.

**Comment by iamhopaul123** [comment]
>I think instead we should provide better and earlier error messages.

Agree. at least we should let them know why the `app init` fails instead of going to the CFN, and how they can fix this problem (like switching region for the `default` profile and do `app delete` again).
